public/app/features/alerting/unified/components/MenuItemPauseRule.tsx

```diff
@@ -33,30 +40,43 @@ const MenuItemPauseRule = ({ rule, onPauseChange }: Props) => {
       return;
     }
     const ruleUid = rule.rulerRule.grafana_alert.uid;
+    const targetGroup = await getGroup({
+      rulerConfig: grafanaRulerConfig,
+      namespace: rule.namespace.uid || rule.rulerRule.grafana_alert.namespace_uid,
+      group: rule.group.name,
+    }).unwrap();
 
-    // Parse the rules into correct format for API
-    const modifiedRules = group.rules.map((groupRule) => {
-      if (!(isGrafanaRulerRule(groupRule.rulerRule) && groupRule.rulerRule.grafana_alert.uid === ruleUid)) {
-        return groupRule.rulerRule!;
-      }
+    if (!targetGroup) {
+      notifyApp.error(
+        `Failed to ${newIsPaused ? 'pause' : 'resume'} the rule. Could not get the target group to update the rule.`
+      );
+      return;
+    }
 
-      return produce(groupRule.rulerRule, (updatedGroupRule) => {
-        updatedGroupRule.grafana_alert.is_paused = newIsPaused;
-      });
-    });
+    // Parse the rules into correct format for API
+    const modifiedRules =
+      targetGroup.rules.map((groupRule) => {
+        if (!(isGrafanaRulerRule(groupRule) && groupRule.grafana_alert.uid === ruleUid)) {
+          return groupRule;
+        }
+        return produce(groupRule, (updatedGroupRule) => {
+          updatedGroupRule.grafana_alert.is_paused = newIsPaused;
+        });
+      }) ?? [];
 
     const payload = {
-      interval: group.interval!,
-      name: group.name,
+      interval: targetGroup.interval! ?? '',
+      name: targetGroup.name ?? '',
       rules: modifiedRules,
     };
 
-    await updateRule({
-      nameSpaceUID: rule.namespace.uid || rule.rulerRule.grafana_alert.namespace_uid,
-      payload,
-    }).unwrap();
+    targetGroup &&
+      (await updateRule({
+        nameSpaceUID: rule.namespace.uid || rule.rulerRule.grafana_alert.namespace_uid,
+        payload,
+      }).unwrap());
 
-    onPauseChange?.();
+    targetGroup && onPauseChange?.();
```

Comment:
You don't need these checks for `targetGroup` as we're now returning early (and don't need the `??` operators either?)

Reply:
truee